shared-bindings/wifi/Radio.c

```diff
@@ -102,6 +101,38 @@ STATIC mp_obj_t wifi_radio_stop_scanning_networks(mp_obj_t self_in) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_stop_scanning_networks_obj, wifi_radio_stop_scanning_networks);
 
+//|     hostname: ReadableBuffer
+//|     """Hostname for wifi interface. When the hostname is altered after interface started/connected
+//|        the changes would only be reflected once the interface restarts/reconnects."""
+//|
+STATIC mp_obj_t wifi_radio_get_hostname(mp_obj_t self_in) {
+    wifi_radio_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return common_hal_wifi_radio_get_hostname(self);
+}
+MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_get_hostname_obj, wifi_radio_get_hostname);
+
+STATIC mp_obj_t wifi_radio_set_hostname(mp_obj_t self_in, mp_obj_t hostname_in) {
+    mp_buffer_info_t hostname;
+    mp_get_buffer_raise(hostname_in, &hostname, MP_BUFFER_READ);
+
+    if (hostname.len < 1 || hostname.len > 63) {
+        mp_raise_ValueError(translate("Hostname must be between 1 and 63 characters"));
```

Comment:
Can this handle unicode? Maybe we should take a string explicitly.

Reply:
DNS hostnames can only be English language letters, digits, and hyphens. I don't know if the ESP32 firmware checks for that.

Reply:
It doesn't look like it's validated: https://github.com/espressif/esp-idf/blob/8bc19ba893e5544d571a753d82b44a84799b94b1/components/esp_netif/lwip/esp_netif_lwip.c#L1108